lib/rucio/client/uploadclient.py

```diff
@@ -166,7 +166,7 @@ def upload(self, items, summary_file_path=None):
             # if register_after_upload, file should be overwritten if it is not registered
             # otherwise if file already exists on RSE we're done
             if register_after_upload:
-                if rsemgr.exists(rse_settings, file_did):
+                if rsemgr.exists(rse_settings, pfn if pfn else file_did):
```

Comment:
Why this change? pfn is preferred, otherwise rsemgr needs to extract pfn itself.